janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/builder/CQLSessionBuilder.java

```diff
@@ -49,50 +54,91 @@ public CqlSession build(Configuration configuration, String[] baseHostnames, int
             contactPoints.add(contactPoint);
         }
 
+        DriverConfigLoader driverConfigLoader;
+        if(configuration.get(CQLConfigOptions.BASE_PROGRAMMATIC_CONFIGURATION_ENABLED)){
+            driverConfigLoader = baseConfigurationLoaderBuilder.build(configuration, contactPoints, baseConnectionTimeoutMS);
+        } else {
+            driverConfigLoader = null;
```

Comment:
I prefer to not initializing variables with unnecessary values. In case we don't initialize value with `null` and later forget to set some value in any of the cases (branches) then the compiler will not compile the code and we can quickly find a mistake. With `null` assignment at the begging we may forget about some case and create a bug later.
I think both approaches are good. One of them reduces a probability of introducing a bug and another improves readability.
I think we could spend time to refactor this code to preserve current logic and improve readability

Reply:
Sounds fair!